build/source/netcdf/def_output.f90

```diff
@@ -326,60 +333,58 @@ subroutine def_variab(ncid,iFreq,spatialDesire,timeDesire,metaData,ivtype,err,me
-   if ((metaData(iVar)%varName=='time').and.(iStat.ne.iLookStat%inst)) cycle
+  ! define statistics index
+  iStat = metaData(iVar)%statIndex(iFreq)
 
-   ! create full variable name
+  ! create full variable name (append statistics info(
+  if(iStat==iLookStat%inst)then
    catName = trim(metaData(iVar)%varName)
-   if (iStat.ne.iLookStat%inst) catName = trim(metaData(iVar)%varName)//'_'//trim(get_statName(iStat))
-
-   ! define variable
-   err = nf90_def_var(ncid,trim(catName),ivtype,dimensionIDs,iVarId)
-   call netcdf_err(err,message); if (err/=0) return
-
-   ! add parameter description
-   catName = trim(metaData(iVar)%vardesc)//' ('//trim(get_statName(iStat))
-   catName = trim(catName)//')'
-   err = nf90_put_att(ncid,iVarId,'long_name',trim(catName))
-   call netcdf_err(err,message); if (err/=0) return
-
-   ! add parameter units
-   catName = trim(metaData(iVar)%varunit)
-   if (iStat==iLookStat%totl) then
-
-    ! make sure that the units of this varaible allow for integration
-    if ((index(catName,'s-1')<=0).and.(index(catName,'s-2')<=0).and.(index(catName,'W m-2')<=0)) then
-     err=20
-     message=trim(message)//'trying to integrate a non-time variable: '//trim(metaData(iVar)%varName)//' - units: '//trim(catName)
-     return
-    endif
-
-    ! change to integrated units
-    if (index(catName,'s-1')>0)       then
-     timePosition = index(catName,'s-1')
-     catName(timePosition:(timePosition+3)) = '   '
-    elseif (index(catName,'s-2')>0)   then
-     timePosition = index(catName,'s-2')
-     catName(timePosition:(timePosition+3)) = 's-1'
-    elseif (index(catName,'W m-2')>0) then
-     timePosition = index(catName,'W')
-     catName(timePosition:(timePosition+1)) = 'J'
-    end if
-
-   end if  ! if an integrated flux
-
-   ! add attribute
-   err = nf90_put_att(ncid,iVarId,'units',trim(catName))
-   call netcdf_err(err,message); if (err/=0) return
-
-   ! add file info to metadata structure
-   metaData(iVar)%ncVarID(iStat) = iVarID
-
-  end do ! looping through statistics
+  else
+   catName = trim(metaData(iVar)%varName)//'_'//trim(get_statName(iStat))
+  endif
+
+  ! define variable
+  err = nf90_def_var(ncid,trim(catName),ivtype,dimensionIDs,iVarId)
+  call netcdf_err(err,message); if (err/=0) return
+
+  ! add parameter description
+  catName = trim(metaData(iVar)%vardesc)//' ('//trim(get_statName(iStat))//')'
+  err = nf90_put_att(ncid,iVarId,'long_name',trim(catName))
+  call netcdf_err(err,message); if (err/=0) return
+
+  ! modify units for the summation
+  catName = trim(metaData(iVar)%varunit)
+  if (iStat==iLookStat%totl) then
+
+   ! make sure that the units of this varaible allow for integration
+   if ((index(catName,'s-1')<=0).and.(index(catName,'s-2')<=0).and.(index(catName,'W m-2')<=0)) then
+    message=trim(message)//'trying to integrate a non-time variable: '//trim(metaData(iVar)%varName)//' - units: '//trim(catName)
+    err=20; return
+   endif
+
+   ! change to integrated units
```

Comment:
nice

Reply:
@greyNearing did this -- just appears here because of a cut-and-paste job

Reply:
Still nice though to actually convert the units.